src/main/java/mate/academy/spring/dao/impl/RoleDaoImpl.java

```diff
@@ -0,0 +1,49 @@
+package mate.academy.spring.dao.impl;
+
+import java.util.Optional;
+import mate.academy.spring.dao.RoleDao;
+import mate.academy.spring.exception.DataProcessingException;
+import mate.academy.spring.model.Role;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+
+public class RoleDaoImpl implements RoleDao {
+    private final SessionFactory factory;
+
+    public RoleDaoImpl(SessionFactory factory) {
+        this.factory = factory;
+    }
+
+    @Override
+    public void add(Role role) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(role);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add new role to DB " + role, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
```

Comment:
You can extends AbstractDao and don't to create new add method
```suggestion
public class RoleDaoImpl extends AbstractDao implements RoleDao {
    private final SessionFactory factory;

    public RoleDaoImpl(SessionFactory factory) {
        this.factory = factory;
    }
```